### ISSUE TITLE
Getting the best out of Mask-RCNN 

### ISSUE THREAD

**Issue by eyildiz-ugoe** [matterport/Mask_RCNN#671]
Hi,

I've been trying to make Mask-RCNN work on my coco-style dataset, however whatever I did, it did not produce any good results. I've started to think on variables that I may have set wrong, but I could not find any. 

Could it be because my training data is too small? I have 18 training images, 2 test images and 2 validation images. Only 2 classes to be detected/segmented (plus the background). Is this too less of data for the task? Doesn't Mask-RCNN use augmentation anyway? Below I am sharing some of the configurations I have:

```
class ComponentsConfig(Config):
    """Configuration for training on the shapes dataset.
    """
    NAME = "components"

    # Train on 1 GPU and 2 images per GPU. Put multiple images on each
    # GPU if the images are small. Batch size is 2 (GPUs * images/GPU).
    GPU_COUNT = 1
   IMAGES_PER_GPU = 1

    # Number of classes (including background)
    NUM_CLASSES = 1 + 2  # background + 2 components (screw and lid)

    # Use smaller images for faster training.
    IMAGE_MAX_DIM = 64
    IMAGE_MIN_DIM = 64

    # Use smaller anchors because our image and objects are small
    RPN_ANCHOR_SCALES = (8, 16, 32, 64, 128)

    # Aim to allow ROI sampling to pick 33% positive ROIs.
    TRAIN_ROIS_PER_IMAGE = 32

    STEPS_PER_EPOCH = 200

    VALIDATION_STEPS = STEPS_PER_EPOCH / 20

```

And this is the training command I am using:

```
 model.train(dataset_train, dataset_validate,
                learning_rate=config.LEARNING_RATE / 10,
                epochs=25,
                layers="all")
```

In short, can you pinpoint to me the most crucial parameters and/or points which I should be careful about? I am trying to recognize screws on HDD surfaces, and I am getting terrible results like these:

![2](https://user-images.githubusercontent.com/35396058/41296856-23c69826-6e5e-11e8-8402-506c73b8fe02.png)
![22](https://user-images.githubusercontent.com/35396058/41296859-24dd38e6-6e5e-11e8-8cc2-922991ec19e0.png)
 

I would really appreciate any idea.

**Comment by baristahell** [comment]
Can you check your annotations and see if they are correctly loaded in your dataset?

**Comment by eyildiz-ugoe** [comment]
@baristahell  I did, and [here](https://github.com/waspinator/deep-learning-explorer/issues/3#issuecomment-396785113) is what I am going through.

**Comment by xiaolisarah** [comment]
i suggest u check the data again,debug it ,i had met the same one ,it turn out the data cannot compare the image

**Comment by eyildiz-ugoe** [comment]
@xiaolisarah Sorry but what do you mean by checking the data? Like what to check there? All images are fine, the extensions are fine. If the loader cannot compare the annotation with the image then it has to be fixed by the authors, hence the ticket.
